store/postgres/src/deployment_store.rs

```diff
@@ -1362,14 +1370,63 @@ impl DeploymentStore {
                 // changes that might need to be reverted
                 Layout::revert_metadata(conn, &site, block)?;
 
-                deployment::update_entity_count(conn, site.as_ref(), count)?;
+                if truncate {
+                    deployment::set_entity_count(conn, site.as_ref(), layout.count_query.as_str())?;
+                } else {
+                    deployment::update_entity_count(conn, site.as_ref(), count)?;
+                }
+
                 Ok(event)
             })
         })?;
 
         Ok(event)
     }
 
+    fn rewind_with_conn(
+        &self,
+        conn: &PgConnection,
+        site: Arc<Site>,
+        block_ptr_to: BlockPtr,
+        firehose_cursor: &FirehoseCursor,
+    ) -> Result<StoreEvent, StoreError> {
+        self.rewind_or_truncate_with_conn(conn, site, block_ptr_to, firehose_cursor, false)
+    }
+
+    fn truncate_with_conn(
+        &self,
+        conn: &PgConnection,
+        site: Arc<Site>,
+        block_ptr_to: BlockPtr,
+        firehose_cursor: &FirehoseCursor,
```

Comment:
Since the firehose cursor will always be set to `None`, it shouldn't be passed in here; this function can just always use `None`